keen/api.py

```diff
@@ -194,12 +183,13 @@ def get_collection(self, event_collection):
 
         return response.json()
 
+    @requires_key(KeenKeys.READ)
```

Comment:
Note that the Python SDK has been checking for Master Key when the Read Key was sufficient here and for get_collection(), so the mismatch is intentional.